app/domain/export/pdf/messages/letter/header.rb

```diff
@@ -43,11 +53,32 @@ def build_address(recipient)
     end
 
     def logo_path
-      Settings.messages.pdf.logo.to_s
+      logo_path_setting(group) || logo_path_setting(group.layer_group)
+    end
+
+    def logo_path_setting(group)
+      group.settings(:messages_letter).picture.path
     end
 
     def sender_address
-      Settings.messages.pdf.address
+      if address_present?(group)
+        group_address(group)
+      elsif address_present?(group.layer_group)
+        group_address(group.layer_group)
+      else
+        ''
+      end
+    end
+
+    def group_address(group)
+      [group.name.to_s.squish,
+       group.address.to_s.squish,
+       [group.zip_code, group.town].compact.join(' ').squish,
+       group.country.to_s.squish].compact.join("\n")
```

Comment:
Auch hier bei `address` `.strip` einfügen. Ich sehe es in der PBS Produktion immer wieder dass die Leute z.B. newlines am Ende des `address` Felds haben, die von einem Copy-Paste übrig geblieben sind.

Reply:
es wird ja #squish verwendet. das inkludiert ja #strip, nicht?